### PR TITLE
Propagate for_compaction from compaction filter

### PR DESCRIPTION
**Pull request by v01dstar**
Solving https://github.com/tikv/titan/issues/324.

This fix only works for 6.29 RocksDB. Once we upgrade RocksDB I will propose a different fix which, instead of propagating `for_compaction`, propagate `rate_limit_priority`.

### CODE REVIEW

File: src/blob_file_cache.h [hunk 1]
```diff
@@ -22,7 +22,12 @@ class BlobFileCache {
   // bytes. The provided buffer is used to store the record data, so
   // the buffer must be valid when the record is used.
   Status Get(const ReadOptions& options, uint64_t file_number,
-             const BlobHandle& handle, BlobRecord* record, OwnedSlice* buffer);
+             const BlobHandle& handle, BlobRecord* record, OwnedSlice* buffer) {
+    return Get(options, file_number, handle, record, buffer, false);
+  }
+  Status Get(const ReadOptions& options, uint64_t file_number,
```
Comment (Connor1996) [inline src/blob_file_cache.h:28]: How about just setting a default value for `for_compaction`

**Review by Connor1996 — COMMENTED** [review]
rest LGTM

**Comment by mzygQAQ** [comment]
I want cherry-pick this

File: src/blob_file_cache.h (hunk 1 above)
Comment (v01dstar) [reply src/blob_file_cache.h]: Done

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Pull request merged**